packages/mcp-server-supabase/src/tools/branching-tools.ts

```diff
@@ -8,11 +8,13 @@ import { injectableTool } from './util.js';
 export type BranchingToolsOptions = {
   platform: SupabasePlatform;
   projectId?: string;
+  readOnly?: boolean;
```

Comment:
We'll need to update `server.ts` to forward the read only setting to all of these new `readOnly` options